x/validator-vesting/client/cli/query.go

```diff
@@ -0,0 +1,48 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/kava-labs/kava/x/validator-vesting/internal/types"
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+// GetQueryValidatorVesting returns the cli query commands for this module
+func GetQueryValidatorVesting(queryRoute string, cdc *codec.Codec) *cobra.Command {
+	// Group nameservice queries under a subcommand
+	queryValidatorVestingCmd := &cobra.Command{
+		Use:   "validator-vesting",
+		Short: "Querying commands for the validator vesting module",
+	}
+
+	queryValidatorVestingCmd.AddCommand(client.GetCommands(
+		QueryCirculatingSupplyCmd(queryRoute, cdc),
+	)...)
+	return queryValidatorVestingCmd
+}
+
+// QueryCirculatingSupplyCmd queries the total circulating supply
+func QueryCirculatingSupplyCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "cirulating-supply",
+		Short: "Query circulating supply information",
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", queryRoute, types.QueryCirculatingSupply), nil)
+			if err != nil {
+				fmt.Printf("could not get total circulating supply\n")
+				return nil
+			}
+
+			var out types.TotalCirculatingSupply
+			cdc.MustUnmarshalJSON(res, &out)
```

Comment:
Looks like your querier returns a json encoded `sdk.Int`, but this will try and unmarshal it into the `TotalCirculatingSupply` type.
You could maybe get away without the type and just unmarshal the querier response into an `sdk.Int`.

Reply:
I fixed this to use `sdk.Int` instead